### DISCUSSION TITLE
How to use wasm file in Foxglove extension?

### DISCUSSION THREAD

**Discussion by RaulCHOU** [foxglove/community#866] [General]
**Background**: use C++ source code in Foxglove extension (typescript)
**Method I used**: use "em++" to build my c++ code, then somehow try to use it
**Status now**: I can make it use in my local node.js environment. But failed in extension when start up foxglove
Local valid code:
```
const fs = require('fs');    
const { WASI } = require('wasi');
     
const wasi = new WASI({
    args: process.argv,
    env: process.env,});
const importObject = { 
    env: {
        env: process.env,
    },
    wasi_snapshot_preview1: wasi.wasiImport
 };    
 (async () => {      
const wasm = await WebAssembly.compile(fs.readFileSync('./wrapper.wasm'));     
 const instance = await WebAssembly.instantiate(wasm, importObject);      
 wasi.start(instance);     
})();
```

For extension tpyescript, I used `WebAssembly.instantiate`, emm... I didn't figure it out.
I also use some other method like FFI (webpackage version gap leads to failure )or Node-Addon-Js (cannot load .node file in web environment, right?). 
Do we have any best practice about this usage? I have blocked by it over weeks. 😮‍💨
Thank a lot in advance.

**Comment by RaulCHOU** [comment]
Otherwise, I can use the .js file which built by em++. But this has another issue that, foxglove would find .wasm in app.asar. 
So how can I specific the searching folder for my wasm file?

**Comment by jtbandes** [comment]
Here is an example of an extension that uses wasm, if that helps: https://github.com/defunctzombie/studio-extension-ogv-example

**Reply by RaulCHOU** [reply]
Thanks for this example, I will look further in it. But it seems not so similar with my case due to ogv has already became a third-party so that I can not know how the wasm file was used in an extension.
